mapper-runtime/src/main/java/com/datastax/oss/driver/internal/mapper/entity/EntityHelperBase.java

```diff
@@ -75,4 +86,70 @@ protected void throwIfKeyspaceMissing() {
               DaoKeyspace.class.getSimpleName()));
     }
   }
+
+  public List<CqlIdentifier> findMissingColumns(
+      List<CqlIdentifier> entityColumns, Collection<ColumnMetadata> cqlColumns) {
+    return findMissingColumnsCql(
+        entityColumns,
+        cqlColumns.stream().map(ColumnMetadata::getName).collect(Collectors.toList()));
+  }
+
+  public List<CqlIdentifier> findMissingColumnsCql(
+      List<CqlIdentifier> entityColumns, Collection<CqlIdentifier> cqlColumns) {
+    List<CqlIdentifier> missingColumns = new ArrayList<>();
+    for (CqlIdentifier entityCqlIdentifier : entityColumns) {
+      if (!cqlColumns.contains(entityCqlIdentifier)) {
+        missingColumns.add(entityCqlIdentifier);
+      }
+    }
+    return missingColumns;
+  }
+
+  /**
+   * When the new instance of a class annotated with {@link Dao} is created an automatic check for
+   * schema validation is performed. It verifies if all {@link Dao} entity fields are present in CQL
+   * table. If not the {@link IllegalArgumentException} exception with detailed message is thrown.
+   * This check has startup overhead so once your app is stable you may want to disable it. The
+   * schema validation check is enabled by default. It can be disabled using the {@link
+   * MapperBuilder#withSchemaValidationEnabled(boolean)} method.
+   */
+  public abstract void validateEntityFields();
+
+  public static List<String> findMissingTypes(
+      Map<CqlIdentifier, GenericType<?>> entityColumns,
+      Map<CqlIdentifier, ColumnMetadata> cqlColumns,
+      CodecRegistry codecRegistry) {
+    List<String> missingCodecs = new ArrayList<>();
+
+    for (Map.Entry<CqlIdentifier, GenericType<?>> entityEntry : entityColumns.entrySet()) {
+      ColumnMetadata columnMetadata = cqlColumns.get(entityEntry.getKey());
+      if (columnMetadata == null) {
+        // this will not happen because it will be catch by the generateMissingColumnsCheck() method
+        throw new IllegalArgumentException(
+            "There is no cql column for entity column: " + entityEntry.getKey());
+      }
+      try {
+        codecRegistry.codecFor(columnMetadata.getType(), entityEntry.getValue());
+      } catch (CodecNotFoundException exception) {
+        missingCodecs.add(
+            String.format(
+                "Field: %s, Entity Type: %s, CQL table type: %s",
+                entityEntry.getKey(), exception.getJavaType(), exception.getCqlType()));
+      }
+    }
+    return missingCodecs;
+  }
+
+  public void throwMissingTypesIfNotEmpty(
+      List<String> missingTypes,
+      CqlIdentifier keyspaceId,
+      CqlIdentifier tableId,
+      String entityClassName) {
+    if (!missingTypes.isEmpty()) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The CQL ks.table: %s.%s defined in the entity class: %s has wrong types:\n%s",
+              keyspaceId, tableId, entityClassName, String.join("\n", missingTypes)));
```

Comment:
"has wrong types" => "declares type mappings that are not supported by the codec registry"